packages/webapp/next.config.js

```diff
@@ -83,26 +83,38 @@ module.exports = {
       env: {
         CURRENT_VERSION: version,
       },
-      rewrites: () => [
-        {
-          source: '/api/sitemaps/:path*',
-          destination: `${process.env.NEXT_PUBLIC_API_URL}/sitemaps/:path*`,
-        },
-        {
-          source: '/search',
-          destination: '/search/:provider',
-          has: [
-            {
-              type: 'query',
-              key: 'provider'
-            }
-          ]
-        },
-        {
-          source: '/search',
-          destination: '/search/posts'
+      rewrites: () => {
+        const rewrites = [
+          {
+            source: '/api/sitemaps/:path*',
+            destination: `${process.env.NEXT_PUBLIC_API_URL}/sitemaps/:path*`,
+          },
+          {
+            source: '/search',
+            destination: '/search/:provider',
+            has: [
+              {
+                type: 'query',
+                key: 'provider'
+              }
+            ]
+          },
+          {
+            source: '/search',
+            destination: '/search/posts'
+          }
+        ];
+
+        // to support GitPod environment and avoid CORS issues, we need to proxy the API requests
+        if (process.env.NEXT_PUBLIC_DOMAIN === 'localhost') {
```

Comment:
This env only exists in gitpod? Need to be 100% sure it doesn't accidentally got set on prod infra 🙈

Reply:
This is also used in prod but with a different value. So we should be safe 🥳 

![Screenshot 2024-10-03 at 2 46 27 PM](https://github.com/user-attachments/assets/f5ce672d-5818-423d-9411-bf4db355b542)